messaging/warriors/contracts/CrossChainWarriors.sol

```diff
@@ -2,78 +2,55 @@
 pragma solidity 0.8.7;
 
 import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
-import "@zetachain/protocol-contracts/contracts/evm/interfaces/ZetaInterfaces.sol";
 import "@zetachain/protocol-contracts/contracts/evm/tools/ZetaInteractor.sol";
+import "@zetachain/protocol-contracts/contracts/evm/interfaces/ZetaInterfaces.sol";
 
 interface CrossChainWarriorsErrors {
     error InvalidMessageType();
-
-    error InvalidTransferCaller();
-
-    error ErrorApprovingZeta();
 }
 
 contract CrossChainWarriors is
-    ERC721("CrossChainWarriors", "CCWAR"),
     ZetaInteractor,
     ZetaReceiver,
-    CrossChainWarriorsErrors
+    CrossChainWarriorsErrors,
+    ERC721("CrossChainWarriors", "CCWAR")
 {
     using Counters for Counters.Counter;
-    using Strings for uint256;
+    bytes32 public constant CROSS_CHAIN_WARRIORS_MESSAGE_TYPE =
+        keccak256("CROSS_CHAIN_CROSS_CHAIN_WARRIORS");
 
-    bytes32 public constant CROSS_CHAIN_TRANSFER_MESSAGE =
-        keccak256("CROSS_CHAIN_TRANSFER");
+    event CrossChainWarriorsEvent(uint256, address, address);
+    event CrossChainWarriorsRevertedEvent(uint256, address, address);
 
+    ZetaTokenConsumer private immutable _zetaConsumer;
     IERC20 internal immutable _zetaToken;
-
-    string public baseURI;
-
     Counters.Counter public tokenIds;
 
-    ZetaTokenConsumer private immutable _zetaConsumer;
-
     constructor(
         address connectorAddress,
         address zetaTokenAddress,
-        address zetaConsumerAddress,
-        bool useEven
+        address zetaConsumerAddress
     ) ZetaInteractor(connectorAddress) {
         _zetaToken = IERC20(zetaTokenAddress);
         _zetaConsumer = ZetaTokenConsumer(zetaConsumerAddress);
 
-        /**
-         * @dev A simple way to prevent collisions between cross-chain token ids
-         * As you can see below, the mint function should increase the counter by two
-         */
         tokenIds.increment();
-        if (useEven) tokenIds.increment();
-    }
-
-    function setBaseURI(string memory baseURIParam) public onlyOwner {
-        baseURI = baseURIParam;
+        tokenIds.increment();
```

Comment:
why?

Reply:
You mean why I've removed `useEven`? As far as I understand it's recommended to use even numbers (and increment by 2), maybe I don't fully understand why we need to have an option to use odd numbering.

Reply:
it's because the example is designed to work on two chains, and we should avoid id collision. then one chain mint even and the other chain the odd

Reply:
I've added odd/even back and modified the deploy script to alternate the parity: e2fb27b

It might not work all that well with 3 or more chains, but I think this is an edge case that can be taken care of in later improvements.

Reply:
Instead of a `bool`, you'd pass an `int` and loop in the contract based on the number of chains you deployed to. Seems like beyond the scope of this tutorial.